demos/demo00_MinimalExample.py

```diff
@@ -4,49 +4,46 @@
 import numpy as np
 import cuqi
 from cuqi.model import LinearModel
-from cuqi.distribution import Gaussian, Cauchy_diff
+from cuqi.distribution import Gaussian, Laplace_diff, Cauchy_diff
 from cuqi.problem import BayesianProblem
 
 # %% Minimal example
 
 # Import data and forward matrix
-A      = np.load("data/Deconvolution.npz")["A"]          #Matrix (numpy)
-b      = np.load("data/Deconvolution.npz")["data"]       #Vector (numpy)
-m,n    = A.shape
+Amat   = np.load("data/Deconvolution.npz")["A"]          #Matrix (numpy)
+y_data = np.load("data/Deconvolution.npz")["data"]       #Vector (numpy)
+m,n    = Amat.shape
 
 # Data from square function
-b = cuqi.testproblem.Deconvolution1D(phantom="square").data
+y_data = cuqi.testproblem.Deconvolution1D(phantom="square").data
 x_exact = cuqi.testproblem.Deconvolution1D(phantom="square").exactSolution
 
 # Set up Bayesian model for inverse problem
-model      = LinearModel(A)                           #Model for inverse problem
-prior      = Gaussian(np.zeros(n),0.1)                #Prior distribution
-likelihood = Gaussian(model,0.05).to_likelihood(b)    #Likelihood function
-IP         = BayesianProblem(likelihood, prior)        #Bayesian model for inverse problem
-IP.UQ(exact=x_exact)                                  #Perform UQ on inverse problem
+A  = LinearModel(Amat)                          # y = Ax. Model for inverse problem
+x  = Gaussian(np.zeros(n), 0.1)                 # x ~ N(0,0.1)
+y  = Gaussian(A@x, 0.05)                        # y ~ N(Ax,0.05)
+IP = BayesianProblem(y, x).set_data(y=y_data)   # Bayesian problem given observed data
+IP.UQ(exact=x_exact)                            # Run UQ analysis
 
 # %%
 # Wrap into CUQI "testproblem".
-TP = cuqi.testproblem.Deconvolution1D(prior=prior)
+TP = cuqi.testproblem.Deconvolution1D(prior=x, phantom="square")
 TP.UQ()
 
 # %%
 # switch prior
 # Set up Bayesian model for inverse problem
-M  = LinearModel(forward=A)                                          #Model for inverse problem
-P  = Cauchy_diff(location=np.zeros(n),scale=0.05,bc_type='neumann')  #Prior distribution
-L  = Gaussian(mean=M,std=0.05).to_likelihood(data=b)                 #Likelihood function
-IP = BayesianProblem(likelihood=L, prior=P)                          #Bayesian model for inverse problem
-IP.UQ(exact=x_exact)                                                 #Perform UQ on inverse problem
+A  = LinearModel(Amat)                                  # y = Ax. Model for inverse problem
+x  = Laplace_diff(np.zeros(n), 0.01, bc_type='zero')    # x ~ Laplace_diff(0,0.01), Zero BC
+y  = Gaussian(A@x, 0.05)                                # y ~ N(Ax,0.05)
+IP = BayesianProblem(y, x).set_data(y=y_data)           # Bayesian problem given observed data
+IP.UQ(exact=x_exact)                                    # Run UQ analysis
```

Comment:
May be can return in samples for illustration 
```suggestion
samples = IP.UQ(exact=x_exact)                            # Run UQ analysis
```